src/locales/index.js

```diff
@@ -1,75 +1,41 @@
-//https://www.w3.org/International/questions/qa-lang-2or3
-// Generally it should be a two or three charaters tag (language) followed by a two/three characters subtag (region), if needed.
-const availableTransations = {en, es}
-
-const mobileTranslations = {
-  en: enMobile,
-  es: esMobile
-}
-
-const defaultLanguage = () => {
-  const polyglot = new Polyglot({onMissingKey: () => null})
-  return extendPolyglot(defaultLocaleTag, polyglot, availableTransations[defaultLocaleTag], mobileTranslations[defaultLocaleTag] )
-}
-
-
-const extendPolyglot = (locale, polyglot, phrases, mobilePhrases) => {
-  polyglot.locale(locale)
-  polyglot.extend(phrases)
-  if (!isDesktop) { polyglot.extend(mobilePhrases) }
-  return polyglot
+type Props = {
+  language: string,
+  children: React.Node,
 }
 
-const findMissingKeys = (defaultKeys, customKeys) => {
-  const newTranslationsSet = new Set(customKeys)
-  const missingKeys =  defaultKeys.filter(element => !newTranslationsSet.has(element))
-  if (missingKeys.length) { console.warn('Missing keys:', missingKeys) }
-}
-
-const polyglotFormatKeys = (phrases) =>
-  Object.keys(new Polyglot({phrases}).phrases)
-
-const verifyKeysPresence = (phrases, polyglot) => {
-  const defaultKeys = Object.keys(polyglot.phrases)
-  const customKeys = polyglotFormatKeys(phrases)
-  findMissingKeys(defaultKeys, customKeys)
-}
+export const LocaleProvider = ({ language, children }: Props) => {
+  const polyglot = initializePolyglot(language)
+  const translate = polyglot.t.bind(polyglot)
+  const parseTranslatedTags = (key, handler) => parseTags(translate(key), handler)
 
-const trySupportedLanguage = (language, polyglot) => {
-  if (availableTransations[language]) {
-    return extendPolyglot(language, polyglot, availableTransations[language], mobileTranslations[language])
-  }
-  console.warn('Locale not supported')
+  return (
+    <LocaleContext.Provider value={{ language, translate, parseTranslatedTags }}>
+      {children}
+    </LocaleContext.Provider>
+  )
 }
 
-const useCustomTranslations = (language, polyglot) => {
-  verifyKeysPresence(language.phrases, polyglot)
-  const newPolyglot = trySupportedLanguage(language.locale, polyglot) || polyglot
-  return extendPolyglot(language.locale, newPolyglot, language.phrases, language.mobilePhrases)
+export type LocalisedType = {
+  translate: (string, ?{}) => string,
+  parseTranslatedTags: (string, { text: string } => React.Node) => React.Node,
+  language: string,
 }
 
-const overrideTranslations = (language, polyglot) => {
-  if (typeof(language) === 'string') {
-    return trySupportedLanguage(language, polyglot)
-  }
-  return useCustomTranslations(language, polyglot)
-}
-
-export const initializeI18n = (language) => {
-  const polyglot = defaultLanguage()
-  if (!language) return polyglot
-  return overrideTranslations(language, polyglot) || polyglot
-}
+export const localised = <WrappedProps: *>(
+  WrappedComponent: React.ComponentType<WrappedProps>
```

Comment:
I find this inline flow syntax quite difficult to read. Can we split it?

Reply:
Sure @stefaniacardenas , how would you split it?

Reply:
```
export const localised = <Props: *>(Wrapped: ComponentType<Props>): ComponentType<{...LocalisedType, ...Props}> =>
  (props: Props) =>
    <LocaleContext.Consumer>{
      (injectedProps: LocalisedType) => <Wrapped {...props} {...injectedProps} />
    }
    </LocaleContext.Consumer>

```

Reply:
Yes, something like that seems slightly more readable. I struggle to read this type of flow syntax but I guess I just need to get used to it 😄